### ISSUE TITLE
EditableField with virtual field

### ISSUE THREAD

**Issue by bluepioupiou** [vitalets/x-editable-yii#2]
Hi, don`t know if it`s a bug but I need to use EditableField on a virtual field.

I want to use this extension to allow people to override a name of an object, but only for their account. So I created a method getCustomName in the class and use customName as attribute in the EditableField. The customName displays but not the clickable link.... but it works fine with name, a 'normal' field
What do I need to do to display this virtual field like a normal one and be able to change it ?

Thanks in advance

**Comment by vitalets** [comment]
hi

for security reason only **safe** attributes become editable (e.g. defined in `rules()`).
Please check what returns `$model->isAttributeSafe()` for virtual attribute?

Another idea (if I understand your task correctly):
use `apply` option to make field editable only for corresponding user. Something like:

```
$this->widget('editable.EditableField', array(
    'type' => 'text',
    'model' => $user,
    'attribute' => 'user_name',
    'url' => $this->createUrl('site/updateUser'),
    'apply' => $user->primaryKey == Yii::app()->user->id,
));
```

**Comment by bluepioupiou** [comment]
Thank you for your answer. I will try the test as soon as possible.

apply is a good thing to know, but it's not what i'm looking for. I want to provide each user a way to 'personnalise' the name of the object.

For the record, I used EditableField on the name field, but when the user enter something, I add the custom name in another table. And I've override the getName method to return the custom name if it exists

**Comment by vitalets** [comment]
it should work in just released 1.1.0
